### ISSUE TITLE
PhishStats feed error

### ISSUE THREAD

**Issue by SteveD3** [ninoseki/miteru#148]
Hello, opening an issue request as mentioned on Twitter. I'm hoping you might be able to assist me.

The problem I'm having is that PhishStats is reporting an error on scan. It's happened over the last few days (error below), but prior 10 Feb 2021, things worked fine.

I get the error when I call miteru via cron (daily at 01:30am) or manually. Also, when I load the PhishStats URL directly (https://phishstats.info:2096/api/phishing?_sort=-id&size=100), things load fine on the browser and it appears to work.

I've rebooted the system. Prior to the errors starting, I made no changes to the server. 

Error: 
02/13/2021
Failed to load PhishStats feed (execution expired)
Loaded 11017 URLs to crawl. (crawling in 12 threads)

02/14/2021
Failed to load PhishStats feed (execution expired)
Loaded 10943 URLs to crawl. (crawling in 12 threads)

02/15/2021
Failed to load PhishStats feed (execution expired)
Loaded 20682 URLs to crawl. (crawling in 12 threads)

Command via cron:

/usr/bin/ruby2.5 /usr/local/bin/miteru --threads=12 --size=10000 --ayashige --auto-download --directory-traveling --download-to=/media/steved3/LINUX/miteru_kit_dl/

Command for manual usage:

miteru command used (manual) miteru --threads=12 --size=10000 --ayashige --auto-download --directory-traveling --download-to=/media/steved3/LINUX/miteru_kit_dl/

OS: Debian 10

steved3@steved3-lab:~$ ruby -v
ruby 2.5.5p157 (2019-03-15 revision 67260) [x86_64-linux-gnu]

miteru v0.14.7

steps taken:
rebooted system
attempted manual scan to see if error persisted (it does)
attempted to load PhishStats URL via browser to confirm API is functional

**Comment by SteveD3** [comment]
Updating this issue... 

Additional research and an accidental (unrelated) change made to my configs today, led me to investigate problems with IPv6 as a possible cause.  As it turns out, there are issues with Ruby and IPv6 when it comes to updates (they'll hang from time to time or just fail), and xrdp has some IPv6 problems too. 

Now, those issues are unrelated to this one, but the symptoms were similar, so I investigated the IPv6 thing.

After testing, it appears my hunch was correct.  I discovered the IPv6 issues while trying to update Ruby and Gems in order to fix my miteru problems. In order to fix the Ruby update issues, I disabled IPv6. Doing so broke my RDP access, but the odd thing was, when IPv6 was disabled, the miteru issues I reported were resolved. 

**OS:** Debian 10

**gem -v**
3.2.10

**ruby -v**
ruby 2.5.8p224 (2020-03-31 revision 67882) [x86_64-linux]


I'm waiting for the cron job that processes miteru to run before I close this issue, but wanted to report my steps and findings in case anyone has these issues in the future. As things stand now, manually running miteru works as expected with no errors. Hopefully, later tonight the cron works too. My fix is below, but this might not work for everyone. It allows me to keep IPv6, but set the order of precedence to IPv4 over IPv6. 

In my research, I came across several useful links, and I've listed them below. 

**Edit:**
/etc/gai.conf

**Enable:**
precedence  ::1/128       50
precedence  ::/0          40
precedence  2002::/16     30
precedence ::/96          20
precedence ::ffff:0:0/96  100

Useful Links:
[Set IPv4 as preferred protocol](https://serverfault.com/questions/93717/setting-ipv4-as-preferred-protocol-over-ipv6)
[gem cannot access rubygems.org](https://stackoverflow.com/questions/49800432/gem-cannot-access-rubygems-org/50349235#50349235)
